Packs/Cofense-Intelligence/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py

```diff
@@ -0,0 +1,390 @@
+        outputs=threats,
+        raw_response=result,
+        readable_output=tableToMarkdown(name=f'Cofense URL Reputation for url {url}', t=md_data,
+                                        headers=['Threat ID', 'Threat Types', 'Verdict', 'Executive Summary',
+                                                 'Campaign', 'Last Published']),
+        indicator=url_indicator,
+
+    )
+
+
+def check_ip_command(client: Client, args: Dict[str, Any], params) -> CommandResults:
+    ip = args.get('ip')
+    if not ip:
+        raise ValueError('IP not specified')
+    try:
+        socket.inet_aton(ip)
+        # legal
+    except socket.error:
+        raise ValueError('Invalid IP')
+
+    # Call the Client function and get the raw response
+    result = client.threat_search_call(ip=ip)
+    threats = result.get('data', {}).get('threats', [])
+    indicator_found = False
+    threshold = params.get('ip_threshold')
+    threshold_score = SEVERITY_SCORE.get(threshold)
+    if not threshold_score:
+        raise Exception(
+            f'Cofense error: Invalid threshold value: {threshold}. Valid values are: None, Minor, Moderate or Major')
+
+    dbot_score = 0
+    dbot_score_obj = Common.DBotScore(indicator=ip, indicator_type=DBotScoreType.IP,
+                                      integration_name=INTEGRATION_NAME, score=dbot_score,
+                                      reliability=params.get(RELIABILITY))
+    adjusted_score = 0
+    ip_indicator = Common.IP(ip=ip, dbot_score=dbot_score_obj)
+    severity_level = 0
+    md_data = []
+    for threat in threats:
+        severity_level = 0
+        for block in threat.get('blockSet'):
+            if block.get('data') == ip and block.get('ipDetail'):
+                ip_indicator.asn = block.get('ipDetail').get('asn')
+                ip_indicator.geo_latitude = block.get("ipDetail").get("latitude")
+                ip_indicator.geo_longitude = block.get("ipDetail").get("longitude")
+                ip_indicator.geo_country = block.get("ipDetail").get("countryIsoCode")
+
+            if block.get('impact'):
+                threat_score = SEVERITY_SCORE.get(block.get('impact'), 0)
+                adjusted_score = 3 if threshold_score <= threat_score else threat_score
+                if block.get('ipDetail') and block.get('ipDetail').get('ip') == ip:
+                    ip_indicator.malware_family = block.get('malwareFamily', {}).get('familyName')
+                    severity_level = adjusted_score
+                    dbot_score = severity_level
+                    indicator_found = True
+                    break
+            severity_level = max(severity_level, adjusted_score)
+        threat_md_row = create_threat_md_row(threat, severity_level)
+        threat_md_row["ASN"] = ip_indicator.asn
+        threat_md_row["Country"] = ip_indicator.geo_country
+        md_data.append(threat_md_row)
+    if not indicator_found:
+        dbot_score = max(dbot_score, severity_level)
```

Comment:
I think this should be separated to a different method to keep this base method shorter and easier to maintain - and this has the same idea as the `threats_analysis` analysis and you separated that one to a different method.

Reply:
Done